src/libcore/panicking.rs

```diff
@@ -36,7 +36,33 @@
                       and related macros",
             issue = "0")]
 
+#[cfg(not(stage0))]
+use any::Any;
 use fmt;
+#[cfg(not(stage0))]
+use panic::{Location, PanicInfo};
+
+#[cfg(not(stage0))]
+#[allow(improper_ctypes)] // PanicInfo contains a trait object which is not FFI safe
+extern "C" {
```

Comment:
Why not `extern "Rust"`? Would that solve the `improper_ctypes` warning?

Reply:
The lint [seems](https://play.rust-lang.org/?gist=cfbcf7185c1f129180142e8264dd3b4d&version=stable&mode=debug) to apply regardless of the ABI ("C" or "Rust").

Reply:
The comment should indicate why it is fine to pass such an object anyway (it never actually passes a FFI boundary and is a Rust-to-Rust call)